src/tools/rpgmodel.h

```diff
@@ -0,0 +1,74 @@
+#ifndef RPGMODEL_H
+#define RPGMODEL_H
+
+#include <QObject>
+#include <QAbstractItemModel>
+#include <data.h>
+
+template <class DATA>
```

Comment:
If you love template programming i recommend taking a look at reader_struct.cpp and reader_struct.h in liblcf >.<
Yeah implementing a specific combobox that is prefilled makes sense, no code duplication. I would have implemented it the same way (probably as a subclass), did this already in another Qt project.